libs/repo-graph/test/suggest/from-one.test.ts

```diff
@@ -25,112 +21,112 @@ const degreeProps: InitProps['Degree'] = {
+    'CS2107',
+    'CS2100',
+    'CS2030',
+    'CP2106',
+    'CS2040C',
+    'CS2040',
+    'CS2030S',
+  ])
+})
+
+it('ranks modules correctly', () => {
+  expect(t.moduleCodes).toEqual([
+    /**
+     * part of degree
+     */
+    'CS2030', // unlocks 4 mods
+    'CS2100', // unlocks 3 mods
+    'CS2107', // unlocks 3 mods
+    'CP2106', // unlocks 0 mods
+    /**
+     * outside of degree
+     */
+    'CS2040', // unlocks 7 mods
+    'CS2040C', // unlocks 2 mods
+    'CS2030S', // unlocks 0 mods
+  ])
+})
+
+test('CS1010 has post-reqs', async () => {
+  await Repo.Module!.findOneByOrFail({ moduleCode: 'CS1010' }).then(
+    (module) => {
+      const codes = module.fulfillRequirements
+      expect(codes).toBeInstanceOf(Array)
+      expect(codes.length).toBeGreaterThan(0)
+      t.postReqsCodes = codes
+    }
+  )
+})
+
+describe('MA3269: not eligible, hence not suggested', () => {
+  it('MA3269 is a post-req of CS1010', () => {
+    expect(t.postReqsCodes).toContain('MA3269')
   })
 
-  describe('Does not suggest post-reqs of the given module', () => {
-    it('Which the user is not eligible for', async () => {
-      // get postReqs
-      const res = await container(db, () =>
-        Repo.Module!.findOneBy({ moduleCode: 'CS1010' })
-      )
-      expect(res).toBeDefined()
-      if (!res) return
-      t.postReqs = res.fulfillRequirements
-      // main test
-      const moduleCodes = ['MA3269', 'DSA3102']
-      // confirm that these modules are indeed CS1010 postReqs
-      moduleCodes.forEach((code) => {
-        expect(t.postReqs!.includes(code)).toEqual(true)
-        expect(t.suggestedModulesCodes!.includes(code)).toEqual(false)
-      })
+  it('user is not eligible for MA3269', async () => {
+    await Repo.User!.canTakeModule(t.user!, 'MA3269').then((res) => {
+      expect(res).toBe(false)
     })
+  })
 
-    it('Which the user has done', async () => {
-      const moduleCodes = ['CG2111A']
-      // confirm that these modules are indeed CS1010 postReqs
-      moduleCodes.forEach((code) => {
-        expect(t.postReqs!.includes(code)).toEqual(true)
-        expect(t.suggestedModulesCodes!.includes(code)).toEqual(false)
-      })
-    })
+  it('MA3269 is not suggested', () => {
+    expect(t.moduleCodes).not.toContain('MA3269')
+  })
+})
+
+describe('DSA3102: not eligible, hence not suggested', () => {
+  it('DSA3102 is a post-req of CS1010', () => {
+    expect(t.postReqsCodes).toContain('DSA3102')
+  })
 
-    it('Which the user is doing', async () => {
-      const moduleCodes = ['IT2002']
-      // confirm that these modules are indeed CS1010 postReqs
-      moduleCodes.forEach((code) => {
-        expect(t.postReqs!.includes(code)).toEqual(true)
-        expect(t.suggestedModulesCodes!.includes(code)).toEqual(false)
-      })
+  it('user is not eligible for DSA3102', async () => {
+    await Repo.User!.canTakeModule(t.user!, 'DSA3102').then((res) => {
+      expect(res).toBe(false)
     })
   })
```

Comment:
i put both MA3269 and DSA3102 at first because it's not expensive, but if we separate the tests then can remove either one of them